index.js

```diff
@@ -1,6 +1,85 @@
-
 import { NativeModules } from 'react-native';
 
 const { RNSnowplowTracker } = NativeModules;
 
-export default RNSnowplowTracker;
+function _applyDefaults(input, defaults) {
+  let combined = input;
+  Object.keys(defaults).forEach((key) => {
+    if(!(key in combined)) {
+      combined[key] = defaults[key];
+      }});
+  return combined;
+}
+
+export default class Tracker {
+
+  static initialize(argmap) {
+
+    let defs = {// defaults for optional params
+                 method: 'post',
+                 protocol: 'https',
+                 setPlatformContext : false,
+                 setBase64Encoded : false,
+                 setApplicationContext : false,
+                 setLifecycleEvents : false,
+                 setScreenContext : false,
+                 setSessionContext : false,
+                 foregroundTimeout : 600,
+                 backgroundTimeout : 300,
+                 checkInterval : 15,
+                 setInstallEvent : false}
+
+
+    if (typeof argmap.endpoint !== 'undefined' && typeof argmap.appId !== 'undefined' && typeof argmap.namespace !== 'undefined' ) {
+      return RNSnowplowTracker.initialize(_applyDefaults(argmap, defs));
+    } else {
+      console.warn("SnowplowTracker: initialize() requires endpoint, namespace and appId parameter to be set")
```

Comment:
I always feel uneasy about console messages, because I feel they add to the bundle size of the package without doing anything helpful.

A clever developer can remove them from the production bundle anyway using webpack options, but I reckon most developers don't do this.

That's just my own thoughts though - I am not suggesting a change here.

Reply:
So my options here felt like they were - throw an error (crashes app), throw a warning (popup appears in debug mode only), or have the native app crash.

Since a) the app crashing is a pain, and b) you don't really want an unexpected edge case to crash your entire app in production, this felt like the option to take.

Are there alternatives to consider here? 

PS. I could also return a promise rejection, but this has the same effect I believe?

Reply:
You could try this:

```
} else if (__DEV__) {
  console.warn("you did something wrong")
}
```

With the implication that if not dev then nothing happens.  Just move on.

The interesting test is whether the react-native script is clever enough to remove this line entirely from a production build (e.g. of the DemoApp).  If it doesn't remove it, then this doesn't really achieve anything - so it would be fine to leave it as you have done it.

Reply:
Ah yeah good point. It seems this may well do what we want it to do: https://stackoverflow.com/questions/55340673/is-there-a-way-to-keep-debug-code-out-of-a-release-build-in-react-native

Nice, thanks for the suggestion!